File: backend/schemas/invoice.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from utils.enums import InvoiceStatus


class InvoiceLineOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    product_name: str
    quantity: float
    unit_price: float
    subtotal: float


class InvoiceOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    invoice_number: str
    sales_order_id: int
    customer_name: str
    customer_email: str | None
    status: InvoiceStatus
    total_amount: float
    generated_at: datetime
    sent_at: datetime | None
    pdf_path: str | None
    lines: list[InvoiceLineOut]

File: frontend/src/types/invoice.ts
export type InvoiceStatus = "DRAFT" | "SENT";

export interface InvoiceLine {
  id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

/** InvoiceOut from GET /invoices. */
export interface Invoice {
  id: number;
  invoice_number: string;
  sales_order_id: number;
  customer_name: string;
  customer_email: string | null;
  status: InvoiceStatus;
  total_amount: number;
  generated_at: string;
  sent_at: string | null;
  pdf_path: string | null;
  lines: InvoiceLine[];
}

/** GET /invoices returns a bare array (matching other list endpoints). */
export type InvoiceListResponse = Invoice[];
